src/components/CouponDetails/helpers.test.jsx

```diff
@@ -36,6 +36,7 @@ describe('getFirstNonDisabledOption', () => {
     [[{ value: 'foo', disabled: true }, { value: 'bar' }], 'bar'],
     [[{ value: 'foo' }, { value: 'bar', disabled: false }, { value: 'baz', disabled: true }], 'foo'],
     [[{ value: 'foo', disabled: true }, { value: 'bar', disabled: true }, { value: 'baz', disabled: false }], 'baz'],
+    [[{ value: 'foo', disabled: false }, { value: 'bar', disabled: true }, { value: 'baz', disabled: true }], 'foo'],
```

Comment:
I just wanted to be extra sure that the logic issue wasn't here, so added a test case that was the same as what I was seeing on stage.